frontend/src/components/pages/secrets/update-secret-modal.tsx

```diff
@@ -102,6 +114,39 @@ export const UpdateSecretModal = ({ isOpen, onClose, secretId }: UpdateSecretMod
                   {(field) => <field.PasswordField label="Value" data-testid="secret-value-field" />}
                 </form.AppField>
 
+                <form.AppField name="scopes">
+                  {({ state, handleChange, handleBlur }) => (
+                    <FormField label="Scopes" errorText=" " isInvalid={state.meta.errors?.length > 0}>
+                      <Select
+                        placeholder="Select scopes"
+                        onChange={(nextValue) => {
+                          if (isMultiValue(nextValue) && nextValue) {
+                            handleChange(nextValue.map(({ value }) => value));
+                          }
+                        }}
+                        options={scopeOptions}
+                        defaultValue={scopeOptions.filter((so) => matchingSecret?.scopes?.some((s) => so.value === s))}
+                        isMulti
+                        onBlur={handleBlur}
+                      />
+                      {
+                        // Display error messages like tanstack/react-form fields.
+                        state?.meta.errors?.length > 0 && (
+                          <FormErrorMessage>
+                            <UnorderedList>
+                              {state.meta.errors?.map((error) => (
+                                <li key={error.path}>
+                                  <Text color="red.500">{error.message}</Text>
+                                </li>
+                              ))}
+                            </UnorderedList>
+                          </FormErrorMessage>
+                        )
+                      }
+                    </FormField>
```

Comment:
You don't need to do this by hand, I think it's enough to do `Form.SingleSelect`, unless we need `MultiSelect` added?

Reply:
This need multiselect option

Reply:
no worries, I answered above, let's move into multi select.